File: src/main/kotlin/record/liquors/liquor/entity/Liquor.kt
package record.liquors.liquor.entity

import record.liquors.base.BaseEntityModel
import java.time.LocalDateTime
import javax.persistence.*
import javax.persistence.FetchType.LAZY

@Entity
class Liquor(
  
  var name: String,
  
  var price: Int,
  
  @ManyToOne(fetch = LAZY)
  @JoinColumn(name = "category_id")
  var category: LiquorCategory,
  
  @OneToMany(fetch = LAZY)
  @JoinColumn(name = "review_id")
  var review: MutableList<Review> = mutableListOf(),
  
  @Id @GeneratedValue
  var id: Long? = null
) : BaseEntityModel() {
  fun update(name: String, price: Int, category: LiquorCategory) {
    this.name = name
    this.price = price
    this.category = category
    this.updatedAt = LocalDateTime.now()
  }
  
  fun addReview(review: Review) {
    this.review.add(review)
  }
}

File: myliquor/src/model/Liquor.ts
import {Category} from './Category';

export class Liquor {
  private readonly _id: number;
  private readonly _name: string;
  private readonly _categoryId: number;
  private readonly _categoryName: string;

  constructor(
    id: number,
    name: string,
    categoryId: number,
    categoryName: string,
  ) {
    this._id = id;
    this._name = name;
    // TODO Category 모델을 참조하게 바꿀 필요가 있어보인다.
    this._categoryId = categoryId;
    this._categoryName = categoryName;
  }

  get id(): number {
    return this._id;
  }

  get name(): string {
    return this._name;
  }

  get categoryId(): number {
    return this._categoryId;
  }

  get categoryName(): string {
    return this._categoryName;
  }
}
